Source/Core/Core/FifoPlayer/FifoDataFile.cpp

```diff
@@ -199,20 +216,56 @@ std::unique_ptr<FifoDataFile> FifoDataFile::Load(const std::string& filename, bo
   if (header.fileId != FILE_ID || header.min_loader_version > VERSION_NUMBER)
   {
     file.Close();
+    ERROR_LOG(BOOT,
+              "The DFF's minimum loader version (%d) exceeds the version of this FIFO Player (%d)",
+              header.min_loader_version, VERSION_NUMBER);
     return nullptr;
   }
 
+  // Official support for overridden RAM sizes was added in version 5.
+  if (header.file_version < 5)
+  {
+    // It's safe to assume FIFO Logs before PR #8722 weren't using this
+    // obscure feature, so load up these header values with the old defaults.
+    header.mem1_size = Memory::MEM1_SIZE_RETAIL;
+    header.mem2_size = Memory::MEM2_SIZE_RETAIL;
+  }
+
   auto dataFile = std::make_unique<FifoDataFile>();
 
   dataFile->m_Flags = header.flags;
   dataFile->m_Version = header.file_version;
 
   if (flagsOnly)
   {
+    // Force settings to match those used when the DFF was created.  This is sort of a hack.
+    // It only works because this function gets called twice, and the first time (flagsOnly mode)
+    // happens to be before HW::Init().  But the convenience is hard to deny!
+    Config::SetCurrent(Config::MAIN_RAM_OVERRIDE_ENABLE, true);
+    Config::SetCurrent(Config::MAIN_MEM1_SIZE, header.mem1_size);
+    Config::SetCurrent(Config::MAIN_MEM2_SIZE, header.mem2_size);
+
     file.Close();
     return dataFile;
   }
 
+  // To make up for such a hacky thing, here is a catch-all failsafe in case if the above code
+  // stops working or is otherwise removed.  As it is, this should never end up running.
+  if (header.mem1_size != Memory::GetRamSizeReal() ||
+      header.mem2_size != Memory::GetExRamSizeReal())
+  {
+    ERROR_LOG(BOOT,
```

Comment:
Due to how serious this is if it happens, I would suggest making this use `PanicAlert` instead of `ERROR_LOG`.

Reply:
Should I similarly change the alert I added for if the min_loader_version is too great?

Reply:
I suppose that makes sense, yes.

Reply:
Maybe even CriticalAlertT (T to make it translatable), so that this is always visible even if the user has disabled panic alerts?